lib/appcanary/configuration.rb

```diff
@@ -0,0 +1,51 @@
+require "yaml"
+
+module Appcanary
+  APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3/monitors"
+
+  class Configuration
+    attr_accessor :base_uri, :api_token, :monitor_name
+
+    def initialize
+      self.base_uri = APPCANARY_DEFAULT_BASE_URI
+    end
+  end
+
+  class ConfigurationError < RuntimeError
+  end
+
+  class << self
+    def configuration
+      @@configuration ||= Configuration.new
```

Comment:
Also, :+1: on immutability - but I think that there should be interfaces which sit comfortably alongside typical ruby/rails patterns. So although I also approve of dependency injection + immutability (and builder patterns if not) in object oriented languages, I suggest we keep the idiomatic rails config as a wrapper for rails degenerates :)